File: LeetCode/C++/169. Majority Element.cpp
class Solution
{
public:
    int majorityElement(vector<int> &nums)
    {
        int count = 0;
        int result = 0;

        for (auto num : nums)
        {
            if (count == 0)
            {
                result = num;
            }

            if (num == result)
            {
                count++;
            }
            else
            {
                count--;
            }
        }

        return result;
    }
};

File: LeetCode/Java/169. Majority Element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 0;
        int majorElement = -1;

        for(int i = 0; i < nums.length; i++) {
            if(count == 0) {
                majorElement = nums[i];
                count++;
            }
            else {
                if(nums[i] == majorElement) {
                    count++;
                }
                else {
                    count--;
                }
            }
        }

        return majorElement;
    }
}